core/eolearn/tests/test_eodata_merge.py

```diff
@@ -186,3 +186,16 @@ def test_lazy_loading(test_eopatch_path):
     assert isinstance(eop.mask.get("CLM"), np.ndarray)
     assert isinstance(eop1.mask.get("CLM"), np.ndarray)
     assert isinstance(eop1.mask_timeless._get_unloaded("LULC"), FeatureIO)  # noqa: SLF001
+
+
+def test_temporally_independent_merge(test_eopatch_path):
+    full_patch = EOPatch.load(test_eopatch_path)
+    part1 = EOPatch.load(test_eopatch_path, temporal_selection=slice(None, 10))
+    part2 = EOPatch.load(test_eopatch_path, temporal_selection=slice(10, None))
+
+    # remove ill-formed data, wrong temporal size
```

Comment:
just to make it a bit more intuitive, since it wasn't clear if it's an issue of the test or from before

```suggestion
    # remove pre-existing ill-formed data of wrong temporal size
```